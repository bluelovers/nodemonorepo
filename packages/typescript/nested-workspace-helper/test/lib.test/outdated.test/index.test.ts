import createSetupTeardown from '../../.lib/setup-teardown'
import { listOutdatedDependencies } from '../../../index'

const { apply } = createSetupTeardown('outdated.yaml')

it('without options', apply(async () => {
  const outdated = await listOutdatedDependencies('root')
  expect(outdated).toMatchSnapshot()
}))

describe('with options', () => {
  it('with beOnInstalled=true', apply(async () => {
    const outdated = await listOutdatedDependencies('root', { baseOnInstalled: true })
    expect(outdated).toMatchSnapshot()
  }))

  it('with custom generateVersionRequirement', apply(async () => {
    const outdated = await listOutdatedDependencies('root', {
      generateVersionRequirement: x => x
    })

    expect(outdated).toMatchSnapshot()
  }))

  it('with custom transformVersionRequirement', apply(async () => {
    const outdated = await listOutdatedDependencies('root', {
      transformVersionRequirement: x => x
    })

    expect(outdated).toMatchSnapshot()
  }))

  it('with all custom options', apply(async () => {
    const outdated = await listOutdatedDependencies('root', {
      baseOnInstalled: true,
      generateVersionRequirement: x => x,
      transformVersionRequirement: x => x
    })

    expect(outdated).toMatchSnapshot()
  }))
})
