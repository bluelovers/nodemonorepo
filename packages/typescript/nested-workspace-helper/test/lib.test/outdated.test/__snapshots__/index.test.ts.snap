// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listOutdatedDependencies with options with all custom options 1`] = `
Array [
  Object {
    "manifestContent": Object {
      "__describe": "All is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "all-out-of-date",
      "version": "987.654.321",
    },
    "manifestFile": "root/allOutOfDate/package.json",
    "path": "root/allOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "bar is up-to-date, foo is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "bar-foo",
      "version": "987.654.321",
    },
    "manifestFile": "root/barUpToDate-fooOutOfDate/package.json",
    "path": "root/barUpToDate-fooOutOfDate",
    "update": Object {
      "dev": Object {},
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "foo is up-to-date, bar is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "foo-bar",
      "version": "987.654.321",
    },
    "manifestFile": "root/fooUpToDate-barOutOfDate/package.json",
    "path": "root/fooUpToDate-barOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {},
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "package.json is up-to-date, node_modules is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "manifest-installed",
      "version": "987.654.321",
    },
    "manifestFile": "root/manifestUpToDate-installedOutOfDate/package.json",
    "path": "root/manifestUpToDate-installedOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement does not cover latest version",
      "dependencies": Object {
        "foo": "1.0.0",
      },
      "devDependencies": Object {
        "bar": "4.0.0",
      },
      "name": "pinned-unsatisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/pinnedUnsatisfy/package.json",
    "path": "root/pinnedUnsatisfy",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
]
`;

exports[`listOutdatedDependencies with options with beOnInstalled=true 1`] = `
Array [
  Object {
    "manifestContent": Object {
      "__describe": "All is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "all-out-of-date",
      "version": "987.654.321",
    },
    "manifestFile": "root/allOutOfDate/package.json",
    "path": "root/allOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "bar is up-to-date, foo is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "bar-foo",
      "version": "987.654.321",
    },
    "manifestFile": "root/barUpToDate-fooOutOfDate/package.json",
    "path": "root/barUpToDate-fooOutOfDate",
    "update": Object {
      "dev": Object {},
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "foo is up-to-date, bar is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "foo-bar",
      "version": "987.654.321",
    },
    "manifestFile": "root/fooUpToDate-barOutOfDate/package.json",
    "path": "root/fooUpToDate-barOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {},
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "package.json is up-to-date, node_modules is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "manifest-installed",
      "version": "987.654.321",
    },
    "manifestFile": "root/manifestUpToDate-installedOutOfDate/package.json",
    "path": "root/manifestUpToDate-installedOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement does not cover latest version",
      "dependencies": Object {
        "foo": "1.0.0",
      },
      "devDependencies": Object {
        "bar": "4.0.0",
      },
      "name": "pinned-unsatisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/pinnedUnsatisfy/package.json",
    "path": "root/pinnedUnsatisfy",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
]
`;

exports[`listOutdatedDependencies with options with custom generateVersionRequirement 1`] = `
Array [
  Object {
    "manifestContent": Object {
      "__describe": "All is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "all-out-of-date",
      "version": "987.654.321",
    },
    "manifestFile": "root/allOutOfDate/package.json",
    "path": "root/allOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "bar is up-to-date, foo is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "bar-foo",
      "version": "987.654.321",
    },
    "manifestFile": "root/barUpToDate-fooOutOfDate/package.json",
    "path": "root/barUpToDate-fooOutOfDate",
    "update": Object {
      "dev": Object {},
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement barely covers latest version",
      "dependencies": Object {
        "foo": "^1.0.0",
      },
      "devDependencies": Object {
        "bar": "^4.0.0",
      },
      "name": "barely-satisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/barelySatisfy/package.json",
    "path": "root/barelySatisfy",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "foo is up-to-date, bar is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "foo-bar",
      "version": "987.654.321",
    },
    "manifestFile": "root/fooUpToDate-barOutOfDate/package.json",
    "path": "root/fooUpToDate-barOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {},
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "node_modules is up-to-date, package.json is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "installed-manifest",
      "version": "987.654.321",
    },
    "manifestFile": "root/installedUpToDate-manifestOutOfDate/package.json",
    "path": "root/installedUpToDate-manifestOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement does not cover latest version",
      "dependencies": Object {
        "foo": "1.0.0",
      },
      "devDependencies": Object {
        "bar": "4.0.0",
      },
      "name": "pinned-unsatisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/pinnedUnsatisfy/package.json",
    "path": "root/pinnedUnsatisfy",
    "update": Object {
      "dev": Object {
        "bar": "4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "1.2.3",
      },
    },
  },
]
`;

exports[`listOutdatedDependencies with options with custom transformVersionRequirement 1`] = `
Array [
  Object {
    "manifestContent": Object {
      "__describe": "All is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "all-out-of-date",
      "version": "987.654.321",
    },
    "manifestFile": "root/allOutOfDate/package.json",
    "path": "root/allOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "bar is up-to-date, foo is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "bar-foo",
      "version": "987.654.321",
    },
    "manifestFile": "root/barUpToDate-fooOutOfDate/package.json",
    "path": "root/barUpToDate-fooOutOfDate",
    "update": Object {
      "dev": Object {},
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "foo is up-to-date, bar is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "foo-bar",
      "version": "987.654.321",
    },
    "manifestFile": "root/fooUpToDate-barOutOfDate/package.json",
    "path": "root/fooUpToDate-barOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {},
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "node_modules is up-to-date, package.json is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "installed-manifest",
      "version": "987.654.321",
    },
    "manifestFile": "root/installedUpToDate-manifestOutOfDate/package.json",
    "path": "root/installedUpToDate-manifestOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement does not cover latest version",
      "dependencies": Object {
        "foo": "1.0.0",
      },
      "devDependencies": Object {
        "bar": "4.0.0",
      },
      "name": "pinned-unsatisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/pinnedUnsatisfy/package.json",
    "path": "root/pinnedUnsatisfy",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
]
`;

exports[`listOutdatedDependencies without options 1`] = `
Array [
  Object {
    "manifestContent": Object {
      "__describe": "All is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "all-out-of-date",
      "version": "987.654.321",
    },
    "manifestFile": "root/allOutOfDate/package.json",
    "path": "root/allOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "bar is up-to-date, foo is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^4.4.4",
      },
      "name": "bar-foo",
      "version": "987.654.321",
    },
    "manifestFile": "root/barUpToDate-fooOutOfDate/package.json",
    "path": "root/barUpToDate-fooOutOfDate",
    "update": Object {
      "dev": Object {},
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement barely covers latest version",
      "dependencies": Object {
        "foo": "^1.0.0",
      },
      "devDependencies": Object {
        "bar": "^4.0.0",
      },
      "name": "barely-satisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/barelySatisfy/package.json",
    "path": "root/barelySatisfy",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "foo is up-to-date, bar is out-of-date",
      "dependencies": Object {
        "foo": "^1.2.3",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "foo-bar",
      "version": "987.654.321",
    },
    "manifestFile": "root/fooUpToDate-barOutOfDate/package.json",
    "path": "root/fooUpToDate-barOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {},
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "node_modules is up-to-date, package.json is out-of-date",
      "dependencies": Object {
        "foo": "^0.1.2",
      },
      "devDependencies": Object {
        "bar": "^3.3.3",
      },
      "name": "installed-manifest",
      "version": "987.654.321",
    },
    "manifestFile": "root/installedUpToDate-manifestOutOfDate/package.json",
    "path": "root/installedUpToDate-manifestOutOfDate",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
  Object {
    "manifestContent": Object {
      "__describe": "Current requirement does not cover latest version",
      "dependencies": Object {
        "foo": "1.0.0",
      },
      "devDependencies": Object {
        "bar": "4.0.0",
      },
      "name": "pinned-unsatisfy",
      "version": "987.654.321",
    },
    "manifestFile": "root/pinnedUnsatisfy/package.json",
    "path": "root/pinnedUnsatisfy",
    "update": Object {
      "dev": Object {
        "bar": "^4.4.4",
      },
      "peer": Object {},
      "prod": Object {
        "foo": "^1.2.3",
      },
    },
  },
]
`;
